packages/sit-onyx/src/components/OnyxDataGrid/features/pagination/pagination.ts

```diff
@@ -1,44 +1,129 @@
+    const scrollContainer = ref<HTMLElement>();
+
+    const { vScrollEnd, isScrollEnd } = useScrollEnd({
+      enabled: computed(() => type === "lazy" && !isLastPage.value && !toValue(options.disabled)),
+      loading,
+    });
+
+    watch(isScrollEnd, (isEnd) => {
+      if (!isEnd) return;
+      state.value.current++;
+    });
+
+    watch(scrollContainer, () => {
+      if (!scrollContainer.value) return;
+      vScrollEnd.mounted(scrollContainer.value);
+    });
 
     return {
       name: PAGINATION_FEATURE,
-      watch: [state, isEnabled, isDisabled, ctx.skeleton],
+      watch: [state, isEnabled, isAsync, isDisabled, ctx.skeleton, isScrollEnd, loading],
       mutation: {
         order: FILTERING_MUTATION_ORDER + 1,
         func: (entries) => {
-          if (!isEnabled.value() || isAsync.value) return entries;
+          if (!isEnabled.value()) return entries;
+
+          let _entries = entries.slice();
+
+          if (!isAsync.value) {
+            state.value.pages = Math.ceil(_entries.length / state.value.pageSize);
+            state.value.current = applyLimits(state.value.current, 1, state.value.pages);
+
+            let startIndex = (state.value.current - 1) * state.value.pageSize;
+            const endIndex = startIndex + state.value.pageSize;
+            if (type !== "select") startIndex = 0;
 
-          state.value.pages = Math.ceil(entries.length / state.value.pageSize);
-          state.value.current = applyLimits(state.value.current, 1, state.value.pages);
+            _entries = _entries.slice(startIndex, endIndex);
+          }
 
-          const startIndex = (state.value.current - 1) * state.value.pageSize;
-          const endIndex = startIndex + state.value.pageSize;
-          return entries.slice(startIndex, endIndex);
+          if (type === "lazy" && loading.value) {
+            _entries.push({
+              id: LAZY_LOADING_ROW_ID,
+              [DataGridRowOptionsSymbol]: {
+                columns: [{ key: "id", type: { name: LAZY_LOADING_TYPE_RENDERER } }],
+              },
+            } satisfies DataGridEntry);
+          }
+
+          if (type === "button" && (!isLastPage.value || loading.value)) {
+            _entries.push({
+              id: BUTTON_LOADING_ROW_ID,
+              [DataGridRowOptionsSymbol]: {
+                columns: [{ key: "id", type: { name: BUTTON_LOADING_TYPE_RENDERER } }],
+              },
+            } satisfies DataGridEntry);
+          }
+
+          return _entries;
         },
       },
+      typeRenderer: {
+        [LAZY_LOADING_TYPE_RENDERER]: createTypeRenderer({
+          cell: {
+            tdAttributes: {
+              class: "onyx-data-grid__lazy-pagination",
+            },
+            component: () => h(OnyxLoadingDots),
+          },
+        }),
+        [BUTTON_LOADING_TYPE_RENDERER]: createTypeRenderer({
+          cell: {
+            tdAttributes: {
+              class: "onyx-data-grid__button-pagination",
+            },
```

Comment:
Prefer to use `colspan`, instead of overwriting the CSS. This ensures a11y and sets the [CSS accordingly](https://github.com/SchwarzIT/onyx/blob/larsrickert/3436-data-grid-button-and-lazy-loading/packages/sit-onyx/src/components/OnyxDataGrid/OnyxDataGridRenderer/OnyxDataGridRenderer.vue#L109). 

```suggestion
          cell: {
            tdAttributes: {
              class: "onyx-data-grid__lazy-pagination",
              colspan: 99,
            },
            component: () => h(OnyxLoadingDots),
          },
        }),
        [BUTTON_LOADING_TYPE_RENDERER]: createTypeRenderer({
          cell: {
            tdAttributes: {
              class: "onyx-data-grid__button-pagination",
              colspan: 99,
            },
```